nvm.sh

```diff
@@ -2724,7 +2724,7 @@ nvm() {
       fi
 
       if nvm_is_version_installed "${VERSION}"; then
-        nvm_err "${VERSION} is already installed."
+        nvm_echo "${VERSION} is already installed."
```

Comment:
Content appearing on stderr doesn't mean things have failed. By doing this, you'll break any tools that rely on parsing stdout.

The primary thing that matters is the exit code. This code path should already exit zero if everything works.

Reply:
Why would anyone parse the output of the install command to detect if a version is already installed? Wouldn't they run a reporting command and parse that? 

Much more common would be invoking the tool as part of CI environment configuration, no? CI system checks for non zero exit code. Dev examines output of the error stream to if something has gone wrong. Dev checks the standard stream to get context if the error log is insufficient. If the error stream gets cluttered with non error messages than so to will the CI logs be cluttered. And that makes it harder to see what's gone wrong.

Why not scrub the script so that only errors go to the error stream. Then, if that's considered breaking, bump the version?

Reply:
If the command exited zero, nothing has gone wrong. If it hasn't, the more information that might help debug the error, the better.

I don't think that would be particularly nice, nor do I think it's an invariant that any CLI provides. stderr is a free-for-all, stdout is intended for consumption by tools, and the exit code is the important bit.